From 8bd5e1d6ba577110d713f92e30fdd73e537ceffc Mon Sep 17 00:00:00 2001
From: Juro Bystricky <juro.bystricky@intel.com>
Date: Fri, 27 Oct 2017 10:43:21 -0700
Subject: mkbuildinf.pl: improve reproducible builds

If the environment contains SOURCE_DATE_EPOCH, generate output such as:

 #define DATE "built on: reproducible build, date unspecified"

instead of the usual, such as:

 #define DATE "built on: Tue Oct 24 17:20:35 2017"

The message is consistent with other messages if SOURCE_DATE_EPOCH
is defined.

Upstream-Status: Submitted [https://github.com/openssl/openssl/pull/4639]

Signed-off-by: Juro Bystricky <juro.bystricky@intel.com>

diff --git a/util/mkbuildinf.pl b/util/mkbuildinf.pl
index 5bf0168..d183078 100755
--- a/util/mkbuildinf.pl
+++ b/util/mkbuildinf.pl
@@ -10,7 +10,12 @@
 my ($cflags, $platform) = @ARGV;
 
 $cflags = "compiler: $cflags";
-$date = localtime();
+if (defined $ENV{'SOURCE_DATE_EPOCH'}) {
+    $date = "reproducible build, date unspecified"
+} else {
+    $date = localtime();
+}
+
 print <<"END_OUTPUT";
 /* auto-generated by util/mkbuildinf.pl for crypto/cversion.c */
 #define CFLAGS cflags
